### PR TITLE
Propagate exceptions of completed actions to launch service main loop

### PR DESCRIPTION
**Pull request by tumtom**
Hi,

exceptions raised by launch actions (e.g. RuntimeError raised by action.DeclareLaunchArgument) are not handled by the "except Exception" block in https://github.com/tumtom/launch/blob/98bcc874fe6f459ade8571697be32c8c74f6b045/launch/launch/launch_service.py#L337.

This is, because the exception is not picked up from the completed Future (https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.exception)

Instead, only the exception handler function in https://github.com/tumtom/launch/blob/98bcc874fe6f459ade8571697be32c8c74f6b045/launch/launch/launch_service.py#L280 is called then, which leads to the following message being printed

```
Task exception was never retrieved
future: <Task finished name='Task-2' coro=<LaunchService._process_one_event() done, defined at ..../opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py:226> exception=RuntimeError("Included launch description missing required argument 'argument (description: 'A description), given: []")>
```

This commit changes the completed action's exception being handled by the "except Exception" block and being printed by the logger as:

`[ERROR] [launch]: Caught exception in launch (see debug for traceback): Included launch description missing required argument 'argument (description: 'A description), given: []`

Signed-off-by: Thomas Rehner <thomas.rehner@aed-engineering.com>

### CODE REVIEW

File: launch/launch/launch_service.py
Comment (hidmic) [inline launch/launch/launch_service.py:329]: @tumtom if more than one exception is thrown, won't we get the same "exception not retrieved" errors?

**Review by hidmic — COMMENTED** [review]
Thanks for your contribution @tumtom ! I only have one question. Then I'll run CI.

File: launch/launch/launch_service.py
Comment (tumtom) [reply launch/launch/launch_service.py:329]: Hi @hidmic
good question. You're probably right. Do you think this could happen and should be handled?
If so, I can think of different options:
- generate a new exception, in which we sum up the exceptions (modifies exception type!)
- log all exceptions that were retrieved, then raise the first (I think this would be better, right?)

The second approach could be implemented like this:
````python
completed_tasks_exceptions = [task.exception() for task in completed_tasks]
completed_tasks_exceptions  = list(filter(None, completed_tasks_exceptions))
if completed_tasks_exceptions:
    self.__logger.error(completed_tasks_exceptions)
    raise completed_tasks_exceptions[0]
````

Comment (hidmic) [reply launch/launch/launch_service.py:329]: @tumtom I'm subjectively inclined towards the second approach. Perhaps we can do better when it comes to logging them, but I'm onboard with the idea.

Comment (tumtom) [reply launch/launch/launch_service.py]: Ok, let me know what you think!
I'm not sure this code will ever execute due to `return_when=asyncio.FIRST_COMPLETED`, but if more than one action would complete and raise, the exceptions won't be lost now

**Review by hidmic — APPROVED** [review]
LGTM pending green CI

CI up to and above `launch`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15761/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10442/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13427/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16013/)

**Comment by hidmic** [comment]
Let's give it another shot:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15779/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10457/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13441/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16033/)

**Comment by hidmic** [comment]
@tumtom `flake8` is still not happy.

**Comment by tumtom** [comment]
Hi @hidmic sorry, I had some troubles executing the linters locally before pushing. Should be fine now, I hope! Do you want me to squash the commits to keep the history clean?

**Comment by hidmic** [comment]
@tumtom third's the charm!

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15839/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10516/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13504/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/16131/)

> Do you want me to squash the commits to keep the history clean?

No need to. We'll squash before merging.

**Comment by hidmic** [comment]
Alright, CI's happy. Going in!

Thank you for your contribution (and patience) @tumtom.

**Pull request merged**
